runtime/generator.go

```diff
@@ -44,6 +44,7 @@ type Generator struct {
 
 // NewGenerator returns a new Generator object that runs the given Block b.
 func NewGenerator(f *Frame, fn func(*Object) (*Object, *BaseException)) *Generator {
+	f.taken = true // Claim the frame from being returned.
```

Comment:
`__frame__()` may no longer be necessary since the traceback module works now. I can't remember the details of why it was necessary off hand.